backend-models/service-tests/showDuggaservice-test.txt

```diff
@@ -0,0 +1,393 @@
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+***
+* Remove inserted row from DB, using hash
+***
+
+	SQL-query DELETE FROM groupdugga WHERE hash = $hash;
+
+—------------------------------------------------------------------------------------------------------------------
+
+—------------------------------------------------------------------------------------------------------------------
+TEST #3 - 111-117
+update active users
+
+***
+Pre-req: $active != NULL
+***
+
+***
+Pre-req variables
+***
+$pre-values =
+    {
+        $hash = “hjk4ert6”
+        $AUtoken = 52
+    }
+
+    SQL-query: INSERT INTO groupdugga(hash,active_users) VALUES($pre-values[0], $pre-values[1])
+
+***
+* Get newly created active users, 
+***
+	SQL-query: SELECT * FROM groupdugga WHERE hash=$pre-values[0]; //
+
+
+***
+* Update information on a active users
+* send parameters input to showDuggaservice.php
+***
+send
+    {
+        $opt = “UPDATEAU”
+        $hash = “hjk4ert6”
+        $newToken = 9999
+    }
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+***
+* Remove inserted row from DB, using hash
+***
+
+	SQL-query: DELETE FROM groupdugga WHERE hash = $hash;
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+***
+Test#4 - 261 - 309
+get data from userAnswer
+***
+
+***
+pre-req: 
+    !isSuperUser($userid) && isset($_SESSION["submission-$courseid-$coursevers-$duggaid-$moment"]) == TRUE &&
+    isset($_SESSION["submission-password-$courseid-$coursevers-$duggaid-$moment"] == TRUE
+***
+
+$pre-values = {
+        $hash = “AcIj-Ld6”
+        $hashpwd = “OjOEh8R”
+        $variant = 13
+    }
+
+    //Get the selected user by hash
+    SELECT password,timesSubmitted,timesAccessed,grade from userAnswer WHERE hash=$hash;
+
+***
+send parameters input to showDuggaservice.php
+***
+send 
+    {
+        $opt = “SAVDU”
+        $hash = “AcIj-Ld6”
+    }
+
+***
+Gather service output
+***
+    Save the values of the echoed array as a JSON (at the end of the file. echo json_encode($array);)
+    This is the expected output for the micro service.
+
+—------------------------------------------------------------------------------------------------------------------
+—------------------------------------------------------------------------------------------------------------------
+TEST #5 - 279 - 287
+update submitted dugga
+
+***
+Pre-req: 
+    !isset($grade) || !$grade > 1 &&
+    isset($dbpwd) && strcmp($hashpwd,$dbpwd) === 0
```

Comment:
we need to make sure that the prerequisites hold, in other words make a manual insert into the database that guaranties that these logical statements hold.